### ISSUE TITLE
Unable to type (AU)

### ISSUE THREAD

**Issue by hensez** [surge-synthesizer/surge#8350]
**Bug Description:**
Surge XT fails to grab keyboard focus from GarageBand. This doesn't happen with other 3rd party AU plugins that accept keyboard input. Surge XT does not receive keyboard inputs, instead, for example, trying to type 'A' into the search field will open the automation track in GarageBand.

**Surge XT Version**

* Version: 1.3.4 / 1.4 nightly
* Plugin Type: [AU]
* Bitness: [64-bit]

**Reproduction Steps:**
1. Load Surge XT in GarageBand
2. Try to use the search function or name a preset.

**Expected Behavior:**
Being able to type.

**Computer Information**
 - OS: Tahoe 26.3.1 (M4)
 - Host: GarageBand
 - Version: 10.4.12 (latest)

**Comment by mkruselj** [comment]
Can you please try the latest nightly build and see if it's working fine for you?

**Comment by hensez** [comment]
Yes, confirmed to be the case on the latest nightly as well. Standalone app working fine of course.

**Comment by baconpaul** [comment]
There's a maddening problem with focus and logic/gb/au and resizing which is hurting a subset of juce plugins. I think it is when plugins repoint keyboard focus explicitly, which we do for accessibility. 

I'm not sure yet of a fix but you aren't crazy and, in some cases, I see this also. But it's neither persistent, reproducible, nor debuggable for me.

It is indeed very irritating. Sometimes close and reopen window solves it.

I'm debating adding an option to 'be not at all accessible' (namely never move keyboard focus) to see if that helps, since most of our screen reader users are standalone or reaper (and a bit live) and not logic.

**Comment by hensez** [comment]
Sounds very maddening indeed. It's highly reproducible on my system, i.e., it has not (yet) worked even once. Regarding the option to turn off accessibility features as a means to fix it: I don't see a downside besides menu clutter.

**Comment by mkruselj** [comment]
u-he introduced keyboard control to their plugins and it's switchable (off by default), so maybe that's the role model here. Except not sure if we want this to be off by default.

**Comment by hensez** [comment]
Okay, after closing and reopening the plugin window 10-ish times it did actually start accepting keyboard input. I suppose it confirms it is indeed the same issue.

**Comment by mkruselj** [comment]
Do we have this issue tracked already, @baconpaul? If so, we should close this one in favor of the older one that is still open.

**Comment by baconpaul** [comment]
I will look this weekend 

The don’t grab focus answer is a menu but some code splat too

**Comment by baconpaul** [comment]
I think #8215 and this are related. Basically we need a mode where we never grab keyboard focus. Tricky because if thats the default the screen reader will show up potentially in the wrong spot. But its gonna be about the best we can do in logic maddeningly

I do notice with 12.2 and tahoe it is a little less terrible also

but add a dont grab and turn that on perhaps by default in logic is i think a solution

tricky. painful.

**Comment by mkruselj** [comment]
I would rather suggest doing what other JUCE plugins do. If they don't have this option just for one host and are instead are doing nothing, so should we. Both options are different flavors of bad UX unfortunately. But I'd err on the side of what everyone else is doing (or not doing) here.

**Comment by baconpaul** [comment]
but then blind users wont be able to use surge. so i would rather not do that.

**Comment by mkruselj** [comment]
I'm a bit confused, sorry. If we never grab focus, how do a11y users ever type anything in?

**Comment by baconpaul** [comment]
We grab focus now.
That breaks logic
I’m proposing in logic we turn off focus grabbing

**Comment by baconpaul** [comment]
Focus grabbing here is ‘moving focus to something other than where you cilck’

**Comment by mkruselj** [comment]
OH! Ok now I get it!

**Comment by baconpaul** [comment]
I'm not gonna get this before I travel but `baconpaul/no-grabfocus` is my starting point which is not mergable. Just reminding myself.
